src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilder.cs

```diff
@@ -248,5 +209,79 @@ public ClusterBuilder Subscribe(IEventSubscriber subscriber)
             _eventAggregator.Subscribe(subscriber);
             return this;
         }
+
+        // private methods
+        private IClusterFactory CreateClusterFactory()
+        {
+            var serverFactory = CreateServerFactory();
+
+            return new ClusterFactory(
+                _clusterSettings,
+                serverFactory,
+                _eventAggregator);
+        }
+
+        private IConnectionPoolFactory CreateConnectionPoolFactory()
+        {
+            var streamFactory = CreateTcpStreamFactory(_tcpStreamSettings);
+
+            var connectionFactory = new BinaryConnectionFactory(
+                _connectionSettings,
+                streamFactory,
+                _eventAggregator);
+
+            return new ExclusiveConnectionPoolFactory(
+                _connectionPoolSettings,
+                connectionFactory,
+                _eventAggregator);
+        }
+
+        private ServerFactory CreateServerFactory()
+        {
+            var connectionPoolFactory = CreateConnectionPoolFactory();
+            var serverMonitorFactory = CreateServerMonitorFactory();
+
+            return new ServerFactory(
+                _clusterSettings.ConnectionMode,
+                _serverSettings,
+                connectionPoolFactory,
+                serverMonitorFactory,
+                _eventAggregator);
+        }
+
+        private IServerMonitorFactory CreateServerMonitorFactory()
+        {
+            var serverMonitorConnectionSettings = _connectionSettings
+                .With(authenticators: new IAuthenticator[] { });
+
+            var serverMonitorTcpStreamSettings = new TcpStreamSettings(_tcpStreamSettings)
+                .With(
+                    readTimeout: _serverSettings.HeartbeatTimeout,
+                    writeTimeout: _serverSettings.HeartbeatTimeout
+                );
+
+            var serverMonitorStreamFactory = CreateTcpStreamFactory(serverMonitorTcpStreamSettings);
```

Comment:
Any way to test that the settings are initialized correctly?

Reply:
Up to you.  At minimum let's test this new piece though.

Reply:
Added test for the new piece.